lib/geoengineer/project.rb

```diff
@@ -12,16 +12,25 @@ class GeoEngineer::Project
   include HasValidations
 
   attr_accessor :org, :name
+
+  ## Environment is an object of type GeoEngineer::Environment
   attr_reader :environment
 
-  validate -> { environments.nil? ? "Project #{full_name} must have an environment" : nil }
+  ## Environments is available from @_attributes generated by HasAttributes as
+  # an Array of environment names ["test","development"...]
+  validate -> { environments.nil? ? "Project #{full_name} must have an environments attribute" : nil }
   validate -> { all_resources.map(&:errors).flatten }
 
   def initialize(org, name, environment, &block)
     @org = org
     @name = name
     @environment = environment
     instance_exec(self, &block) if block_given?
+    ## environments is an attribute that needs to be set, so lets make sure to set it.
+    if !@environment.nil? &&
+       !@environment.name
```

Comment:
You can use the safe navigation operator here as well:

```ruby
if environment&.name
```

Reply:
Resolved